tensorflow/contrib/distributions/python/ops/beta.py

```diff
@@ -292,3 +293,38 @@ def __init__(self,
           validate_args=validate_args,
           allow_nan_stats=allow_nan_stats,
           name=ns)
+
+
+def _kl_beta_beta(d1, d2, name=None):
+  """Calculate the batched KL divergence KL(d1 || d2) with d1 and d2 Beta.
+
+  Args:
+    d1: instance of a Beta distribution object.
+    d2: instance of a Beta distribution object.
+    name: (optional) Name to use for created operations.
+      default is "kl_beta_beta".
+
+  Returns:
+    Batchwise KL(d1 || d2)
+  """
+  inputs = [d1.a, d1.b, d1.a_b_sum, d2.a_b_sum]
+  with ops.name_scope(
+    name, "kl_beta_beta", inputs):
+      log_betas = (math_ops.lgamma(d2.a) + math_ops.lgamma(d2.b)
+                 - math_ops.lgamma(d2.a_b_sum) + math_ops.lgamma(d1.a_b_sum)
+                 - math_ops.lgamma(d1.a) - math_ops.lgamma(d1.b))
+      digammas = ((d1.a - d2.a)*math_ops.digamma(d1.a)
```

Comment:
Write in a comment above digammas:

(a - a') * psi(a) + (b - b') * psi(b) + (a' + b' - a - b) * psi(a + b)